File: src/main/java/cz/vhromada/catalog/mapper/SeasonMapper.java
package cz.vhromada.catalog.mapper;

import cz.vhromada.catalog.entity.Season;

import org.mapstruct.Mapper;
import org.mapstruct.Mapping;

/**
 * An interface represents mapper for season.
 *
 * @author Vladimir Hromada
 */
@Mapper
public interface SeasonMapper {

    /**
     * Maps entity season to domain season.
     *
     * @param source entity season
     * @return mapped domain season
     */
    @Mapping(target = "episodes", ignore = true)
    cz.vhromada.catalog.domain.Season map(Season source);

    /**
     * Maps domain season to entity season.
     *
     * @param source domain season
     * @return mapped entity season
     */
    Season mapBack(cz.vhromada.catalog.domain.Season source);

}

File: src/main/kotlin/cz/vhromada/catalog/mapper/SeasonMapper.kt
package cz.vhromada.catalog.mapper

import cz.vhromada.catalog.entity.Season
import cz.vhromada.common.mapper.Mapper
import org.springframework.stereotype.Component

/**
 * A class represents mapper for season.
 *
 * @author Vladimir Hromada
 */
@Component("seasonMapper")
class SeasonMapper : Mapper<Season, cz.vhromada.catalog.domain.Season> {

    override fun map(source: Season): cz.vhromada.catalog.domain.Season {
        return cz.vhromada.catalog.domain.Season(
                id = source.id,
                number = source.number!!,
                startYear = source.startYear!!,
                endYear = source.endYear!!,
                language = source.language!!,
                subtitles = source.subtitles!!.filterNotNull(),
                note = source.note,
                position = source.position,
                episodes = emptyList(),
                audit = null)
    }

    override fun mapBack(source: cz.vhromada.catalog.domain.Season): Season {
        return Season(
                id = source.id,
                number = source.number,
                startYear = source.startYear,
                endYear = source.endYear,
                language = source.language,
                subtitles = source.subtitles,
                note = source.note,
                position = source.position)
    }

}
